File: core/src/com/mirage/view/scene/objects/StaticTexture.java
package com.mirage.view.scene.objects;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;

/**
 * Объект без анимации, заданный статичной текстурой
 */
public class StaticTexture extends ObjectDrawer{
    private Texture texture;

    public StaticTexture(Texture texture) {
        this.texture = texture;
    }

    @Override
    public void draw(SpriteBatch batch, float x, float y) {
        batch.draw(texture, x, y);
    }
}

File: core/src/com/mirage/view/scene/objects/StaticTexture.kt
package com.mirage.view.scene.objects

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch

/**
 * Объект без анимации, заданный статичной текстурой
 */
class StaticTexture(private val texture: Texture) : ObjectDrawer() {

    override fun draw(batch: SpriteBatch, x: Float, y: Float) {
        batch.draw(texture, x, y)
    }
}
